lib/pbench/agent/tool_meister.py

```diff
@@ -1518,6 +1539,35 @@ def end_tools(self, data):
                     name,
                 )
                 failures += 1
+
+        # Remove persistent tool temporary working directory
+        directory = data["directory"]
+        tool_dir = self.directories[directory]
+
+        self.logger.debug(
+            "%s: deleting persistent tool tmp directory %s %s",
+            self._hostname,
+            self._group,
+            tool_dir,
+        )
+        unexpected_files = []
+        for dir, _, files in os.walk(tool_dir):
+            dirpath = Path(dir).relative_to(tool_dir)
+            if files:
+                relative_file_path = map(lambda x: f"{dirpath}/{x}", files)
+                unexpected_files += relative_file_path
+
+        if unexpected_files:
+            self.logger.warning(
+                f"{self._hostname}: unexpected temp files {','.join(unexpected_files)}"
+            )
+        try:
+            shutil.rmtree(tool_dir)
+        except Exception:
+            self.logger.exception(
+                f"Failed to remove persistent tool data tmp directory: " f"{tool_dir}"
```

Comment:
And again ... `"Failed to remove persistent tool data tmp directory: %s", tool_dir`.